Library/Homebrew/utils/curl.rb

```diff
@@ -19,6 +20,15 @@ def curl(*args)
 end
 
 def curl_output(*args)
-  curl_args = curl_args(args) - ["--fail"]
-  Utils.popen_read_text(*curl_args)
+  curl_args = curl_args(args)
+  curl_args -= ["--fail"]
+  if RUBY_TWO
+    curl_args -= ["--silent"]
+    Open3.popen3(*curl_args) do |_, stdout, stderr, status|
+      [stdout.read, stderr.read, status]
```

Comment:
Turns out, sometimes you have to check my suggestions. The fourth argument to the block is not actually a status but the associated wait thread, thus this part should be:

```ruby
Open3.popen3(*curl_args) do |_, stdout, stderr, wait_thread|
  [stdout.read, stderr.read, wait_thread.value]
end
```

Otherwise you'll get this error (e.g. when the rate limit hits):

```console
$ brew search thing
syncthing                                                    syncthing-inotify
Error: undefined method `success?' for #<Thread:0x007fca8387d9f8 dead>
Please report this bug:
    https://git.io/brew-troubleshooting
/opt/brewery/tests/Library/Homebrew/utils/github.rb:163:in `open'
/opt/brewery/tests/Library/Homebrew/cmd/search.rb:133:in `block in search_tap'
/opt/brewery/tests/Library/Homebrew/cmd/search.rb:150:in `yield'
/opt/brewery/tests/Library/Homebrew/cmd/search.rb:150:in `search_tap'
/opt/brewery/tests/Library/Homebrew/cmd/search.rb:117:in `block (2 levels) in search_taps'
```